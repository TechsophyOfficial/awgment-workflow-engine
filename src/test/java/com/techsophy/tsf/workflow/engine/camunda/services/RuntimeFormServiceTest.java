//package com.techsophy.tsf.workflow.engine.camunda.services;
//
//import au.com.origin.snapshots.junit5.SnapshotExtension;
//import com.fasterxml.jackson.databind.ObjectMapper;
//import com.github.tomakehurst.wiremock.WireMockServer;
//import com.github.tomakehurst.wiremock.client.ResponseDefinitionBuilder;
//import com.techsophy.tsf.workflow.engine.camunda.config.TestSecurityConfig;
//import com.techsophy.tsf.workflow.engine.camunda.dto.FormSchema;
//import com.techsophy.tsf.workflow.engine.camunda.service.impl.RuntimeFormServiceImpl;
//import lombok.AllArgsConstructor;
//import lombok.Cleanup;
//import org.junit.jupiter.api.*;
//import org.junit.jupiter.api.extension.ExtendWith;
//import org.springframework.beans.factory.annotation.Autowired;
//import org.springframework.boot.test.context.SpringBootTest;
//import org.springframework.core.io.ClassPathResource;
//import org.springframework.http.MediaType;
//import org.springframework.test.context.ActiveProfiles;
//
//import java.io.IOException;
//import java.io.InputStream;
//import java.util.Map;
//
//import static au.com.origin.snapshots.SnapshotMatcher.expect;
//import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
//import static com.github.tomakehurst.wiremock.client.WireMock.get;
//import static com.techsophy.tsf.workflow.engine.camunda.constants.FormConstants.FORM_KEY;
//import static com.techsophy.tsf.workflow.engine.camunda.constants.FormConstants.FORM_NAME;
//import static org.assertj.core.api.Assertions.assertThat;
//
//
//@ActiveProfiles("test")
//@TestInstance(TestInstance.Lifecycle.PER_CLASS)
//@ExtendWith(SnapshotExtension.class)
//@AllArgsConstructor(onConstructor_ = {@Autowired})
//@SpringBootTest(classes = TestSecurityConfig.class)
//@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
//class RuntimeFormServiceTest
//{
//    private static final String FORM_CONTENT_DATA = "testdata/form-content.json";
//
//    private final RuntimeFormServiceImpl formServiceTest;
////    private final RuntimeFormRepository runtimeForm;
//    private final ObjectMapper objectMapper;
//
//    private static final WireMockServer GATEWAY_URL=new WireMockServer(6666);
//
//    @BeforeAll void init() throws IOException {
//        GATEWAY_URL.start();
//        @Cleanup InputStream contactDetails = new ClassPathResource(FORM_CONTENT_DATA).getInputStream();
//        FormSchema form = this.objectMapper.readValue(contactDetails, FormSchema.class);
//        GATEWAY_URL.stubFor(get("/form-runtime/v1/forms/FormKeyTest")
//                .willReturn(json(objectMapper.writeValueAsString(Map.of(
//                        "data",form,
//                        "success",true,
//                        "message","Form details detched successfully")))));
//    }
//
//    public static ResponseDefinitionBuilder json(String apiResponse)
//    {
//        return aResponse().withHeader("Content-Type", MediaType.APPLICATION_JSON_VALUE).withBody(apiResponse);
//    }
//
//    @AfterAll
//    void destroy(){
//        GATEWAY_URL.resetAll();
//        GATEWAY_URL.stop();
//    }
//
//
//
//    @Test
//    @Order(1)
//    void getFormTest()
//    {
//        FormSchema formDetails = formServiceTest.fetchFormById(FORM_KEY,null);
//        assertThat(formDetails.getId()).isEqualTo(FORM_KEY);
//        assertThat(formDetails.getName()).isEqualTo(FORM_NAME);
//        assertThat(formDetails.getComponents()).isNotNull();
//
//        expect(formDetails).orderedJson().toMatchSnapshot();
//    }
//
//
//
//
//}
