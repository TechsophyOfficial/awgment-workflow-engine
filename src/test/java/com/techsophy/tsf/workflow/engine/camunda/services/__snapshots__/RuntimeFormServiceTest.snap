com.techsophy.tsf.workflow.engine.camunda.services.RuntimeFormServiceTest.getFormTest=[
  {
    "components": {
      "components": [
        {
          "input": true,
          "key": "textField",
          "label": "Text Field",
          "tableView": true,
          "type": "textfield"
        },
        {
          "autoExpand": false,
          "input": true,
          "key": "textArea",
          "label": "Text Area",
          "tableView": true,
          "type": "textarea"
        },
        {
          "input": true,
          "inputType": "checkbox",
          "key": "selectBoxes",
          "label": "Select Boxes",
          "optionsLabelPosition": "right",
          "tableView": false,
          "type": "selectboxes",
          "values": [
            {
              "label": "s",
              "shortcut": "A",
              "value": "s"
            }
          ]
        },
        {
          "input": true,
          "key": "email",
          "label": "Email",
          "tableView": true,
          "type": "email"
        },
        {
          "datePicker": {
            "disableWeekdays": false,
            "disableWeekends": false
          },
          "enableMaxDateInput": false,
          "enableMinDateInput": false,
          "input": true,
          "key": "dateTime",
          "label": "Date / Time",
          "tableView": false,
          "type": "datetime",
          "widget": {
            "allowInput": true,
            "disableWeekdays": false,
            "disableWeekends": false,
            "displayInTimezone": "viewer",
            "enableTime": true,
            "format": "yyyy-MM-dd hh:mm a",
            "hourIncrement": 1,
            "locale": "en",
            "minuteIncrement": 1,
            "mode": "single",
            "noCalendar": false,
            "time_24hr": false,
            "type": "calendar",
            "useLocaleSettings": false
          }
        },
        {
          "delimiter": false,
          "input": true,
          "inputFormat": "plain",
          "key": "number",
          "label": "Number",
          "mask": false,
          "requireDecimal": false,
          "spellcheck": true,
          "tableView": false,
          "type": "number"
        },
        {
          "input": true,
          "key": "phoneNumber",
          "label": "Phone Number",
          "tableView": true,
          "type": "phoneNumber"
        },
        {
          "input": true,
          "key": "checkbox",
          "label": "Checkbox",
          "tableView": false,
          "type": "checkbox"
        },
        {
          "action": "next",
          "input": true,
          "key": "next",
          "label": "next",
          "showValidations": false,
          "tableView": false,
          "type": "button"
        },
        {
          "disableOnInvalid": true,
          "input": true,
          "key": "submit",
          "label": "Submit",
          "tableView": false,
          "type": "button"
        }
      ],
      "display": "form"
    },
    "id": "FormKeyTest",
    "name": "FormTest",
    "version": 1
  }
]